torchvision/ops/poolers.py

```diff
@@ -153,14 +169,18 @@ def forward(self, x, boxes, image_shapes):
             device=device,
         )
 
+        results = []
         for level, (per_level_feature, scale) in enumerate(zip(x, self.scales)):
             idx_in_level = torch.nonzero(levels == level).squeeze(1)
             rois_per_level = rois[idx_in_level]
-
-            result[idx_in_level] = roi_align(
+            result_idx_in_level = roi_align(
                 per_level_feature, rois_per_level,
                 output_size=self.output_size,
-                spatial_scale=scale, sampling_ratio=self.sampling_ratio
-            )
+                spatial_scale=scale, sampling_ratio=self.sampling_ratio)
 
+            if torch._C._get_tracing_state():
```

Comment:
How many places you expect that we will need to use this private method?
If this is going to be used fairly often, I'm thinking if we shouldn't have a private method in torchvision (like `torchvision._is_tracing()` that dispatches to `torch._C._get_tracing_state()` for now.

This will make it easier to replace the code with a public API (from PyTorch) afterwards, for example a variant of https://github.com/pytorch/pytorch/pull/25263 which supports tracing.

Thoughts?